stdlib/Test/src/Test.jl

```diff
@@ -64,17 +64,18 @@ function test_callsite(bt, file_ts, file_t)
     # For that, we retrieve locations from lower to higher stack elements
     # and only traverse parts of the backtrace which we haven't traversed before.
     # The order will always be <internal functions> -> `@test` -> `@testset`.
-    internal = macrocall_location(bt, @__FILE__)::Int
+    internal = @something(macrocall_location(bt, @__FILE__), return nothing)
     test = internal - 1 + findfirst(ip -> any(frame -> in_file(frame, file_t), StackTraces.lookup(ip)), @view bt[internal:end])::Int
```

Comment:
We also need to change this line to fix the following error?
```julia
julia> foo() = @test false
foo (generic function with 1 method)

julia> @testset "foo" begin
           foo()
       end
foo: Test Failed at REPL[18]:1
  Expression: false
foo: Error During Test at REPL[19]:1
  Got exception outside of a @test
  TypeError: in typeassert, expected Int64, got a value of type Nothing
  Stacktrace:
    [1] test_callsite(bt::Vector{Union{Ptr{Nothing}, Base.InterpreterIP}}, file_ts::String, file_t::String)
      @ Test ~/julia/julia/usr/share/julia/stdlib/v1.10/Test/src/Test.jl:68
    [2] test_location(bt::Vector{Union{Ptr{Nothing}, Base.InterpreterIP}}, file_ts::String, file_t::String)
      @ Test ~/julia/julia/usr/share/julia/stdlib/v1.10/Test/src/Test.jl:58 [inlined]
    [3] scrub_backtrace(bt::Vector{Union{Ptr{Nothing}, Base.InterpreterIP}}, file_ts::String, file_t::String)
      @ Test ~/julia/julia/usr/share/julia/stdlib/v1.10/Test/src/Test.jl:94
    [4] record(ts::Test.DefaultTestSet, t::Union{Test.Error, Test.Fail}; print_result::Bool)
      @ Test ~/julia/julia/usr/share/julia/stdlib/v1.10/Test/src/Test.jl:1084
    [5] record
      @ Test ~/julia/julia/usr/share/julia/stdlib/v1.10/Test/src/Test.jl:1077 [inlined]
    [6] do_test(result::Test.ExecutionResult, orig_expr::Any)
      @ Test ~/julia/julia/usr/share/julia/stdlib/v1.10/Test/src/Test.jl:704
    [7] macro expansion
      @ Main ~/julia/julia/usr/share/julia/stdlib/v1.10/Test/src/Test.jl:512 [inlined]
    [8] foo()
      @ Main ./REPL[18]:1
    [9] macro expansion
      @ REPL[19]:2 [inlined]
   [10] macro expansion
      @ ~/julia/julia/usr/share/julia/stdlib/v1.10/Test/src/Test.jl:1546 [inlined]
   [11] top-level scope
      @ REPL[19]:2
   [12] eval(m::Module, e::Any)
      @ Core ./boot.jl:378 [inlined]
   [13] eval_user_input(ast::Any, backend::REPL.REPLBackend, mod::Module)
      @ REPL ~/julia/julia/usr/share/julia/stdlib/v1.10/REPL/src/REPL.jl:150
   [14] repl_backend_loop(backend::REPL.REPLBackend, get_module::Function)
      @ REPL ~/julia/julia/usr/share/julia/stdlib/v1.10/REPL/src/REPL.jl:246
   [15] start_repl_backend(backend::REPL.REPLBackend, consumer::Any; get_module::Function)
      @ REPL ~/julia/julia/usr/share/julia/stdlib/v1.10/REPL/src/REPL.jl:231
   [16] run_repl(repl::REPL.AbstractREPL, consumer::Any; backend_on_current_task::Bool, backend::Any)
      @ REPL ~/julia/julia/usr/share/julia/stdlib/v1.10/REPL/src/REPL.jl:376
   [17] run_repl(repl::REPL.AbstractREPL, consumer::Any)
      @ REPL ~/julia/julia/usr/share/julia/stdlib/v1.10/REPL/src/REPL.jl:362
   [18] (::Base.var"#992#994"{Bool, Bool, Bool})(REPL::Module)
      @ Base ./client.jl:421
   [19] invokelatest(::typeof(Base.CoreLogging.handle_message), ::Base.CoreLogging.AbstractLogger, ::Base.CoreLogging.LogLevel, ::String, ::Module, ::Symbol, ::Symbol, ::String, ::Int64; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, @NamedTuple{}})
      @ Base ./essentials.jl:863 [inlined]
   [20] invokelatest(f::Base.var"#992#994"{Bool, Bool, Bool}, args::Module)
      @ Base ./essentials.jl:860 [inlined]
   [21] run_main_repl(interactive::Bool, quiet::Bool, banner::Bool, history_file::Bool, color_set::Bool)
      @ Base ./client.jl:405
   [22] exec_options(opts::Base.JLOptions)
      @ Base ./client.jl:322
   [23] _start()
      @ Base ./client.jl:541
Test Summary: | Error  Total  Time
foo           |     1      1  0.0s
ERROR: Some tests did not pass: 0 passed, 0 failed, 1 errored, 0 broken.
```

Reply:
Sigh, I missed this typeassert. Thanks for checking!

Reply:
https://github.com/JuliaLang/julia/pull/49646